woocommerce/woocommerce-ios
# Fix-4579: Wrong product price in Order Detail if the price of the product is negative

Fixes #4579 

## Description

Currently the `Order View`, `Order Details View`, and `Refund View` show a positive price number when a negative price is set for the product.

As per this [open issue](https://github.com/woocommerce/woocommerce/issues/25037) we want to allow negative numbers, mainly due to plugins that rely on this in order to function properly, meaning that showing a positive or a zero price in the UI when this is actually negative is incorrect.

## Changes

The problem comes from the Order initializer in `ProductDetailsCellViewModel.swift`, where we're setting `positiveTotal`, and `positivePrice` parameters to be absolute-numbers-only via the `abs()` method:

- Removed the `abs()` method from these parameters.
- Renamed the parameters from `positiveTotal` to `total`, and from `positivePrice` to `price`

The price shows as expected now:

<img width="1055" alt="fixed-4579 Order details view" src="https://user-images.githubusercontent.com/3812076/127256105-d20e32db-30cc-4336-81c8-02f5df3f8470.png">

## Testing steps
1. A negative-priced product can be created via the website, the app currently does not allow to create negative-priced products.
2. Fill an order with negative-priced product/s
3. From the app: Tap on Orders > Order > See Order details

## Work in progress / Questions:

- [x] At the moment is not possible to add a product with a negative price via the app, only the web. Created new bug report here: https://github.com/woocommerce/woocommerce-ios/issues/4692

Update release notes:
